crates/gitbutler-repo/src/rebase.rs

```diff
@@ -225,11 +227,84 @@ fn commit_conflicted_cherry_result<'repository>(
         .context("failed to find commit")
 }
 
+fn extract_conflicted_files(
+    merged_tree_id: gix::Id<'_>,
+    merge_result: gix::merge::tree::Outcome<'_>,
+    treat_as_unresolved: gix::merge::tree::TreatAsUnresolved,
+) -> Result<ConflictEntries> {
+    use gix::index::entry::Stage;
+    let repo = merged_tree_id.repo;
+    let mut index = repo.index_from_tree(&merged_tree_id)?;
+    merge_result.index_changed_after_applying_conflicts(
+        &mut index,
+        treat_as_unresolved,
+        gix::merge::tree::apply_index_entries::RemovalMode::Mark,
+    );
+    let (mut ancestor_entries, mut our_entries, mut their_entries) =
+        (Vec::new(), Vec::new(), Vec::new());
+    let conflicting_entries = index.entries().iter().filter_map(|e| {
+        let stage = e.stage();
+        if stage == gix::index::entry::Stage::Unconflicted {
+            None
+        } else {
+            Some((stage, e))
+        }
+    });
+    for (stage, entry) in conflicting_entries {
+        let storage = match stage {
+            Stage::Unconflicted => {
+                unreachable!("BUG: filtered above to not contain unconflicted entries")
+            }
+            Stage::Base => &mut ancestor_entries,
+            Stage::Ours => &mut our_entries,
+            Stage::Theirs => &mut their_entries,
+        };
+
+        let path = entry.path(&index);
+        storage.push(gix::path::from_bstr(path).into_owned());
+    }
+    let mut out = ConflictEntries {
+        ancestor_entries,
+        our_entries,
+        their_entries,
+    };
+
+    // Since we typically auto-resolve with 'ours', it maybe that conflicting entries don't have an
+    // unconflicting counterpart anymore, so they are not applied (which is also what Git does).
+    // So to have something to show for - we *must* produce a conflict, extract paths manually.
+    // TODO(ST): instead of doing this, don't pre-record the paths. Instead redo the merge without
+    //           merge-strategy so that the index entries can be used instead.
```

Comment:
Long term, this would be a good thing to do.

We were previously eagerly loading the diffs of all commits, but this is no longer the case.

Reply:
Indeed, I also think that the new sidebar-branch implementation shows how this can work. And I am looking forward to one day see all this lazy evaluation realized.